include/parallel/parallel_sync.h

```diff
@@ -309,6 +337,182 @@ void push_parallel_vector_data(const Communicator & comm,
 }
 
 
+template <typename MapToVectors,
+          typename ActionFunctor,
+          typename ReceiveContext>
+void push_parallel_packed_range(const Communicator & comm,
+                                const MapToVectors & data,
+                                ReceiveContext * receive_context,
+                                const ActionFunctor & act_on_data)
+{
+  // This function must be run on all processors at once
+  libmesh_parallel_only(comm);
+
+  // This function implements the "NBX" algorithm from
+  // https://htor.inf.ethz.ch/publications/img/hoefler-dsde-protocols.pdf
+
+  // This function only works for "flat" data that we can pre-size
+  // receive buffers for: a map to vectors-of-standard-types, not e.g.
+  // vectors-of-vectors.
```

Comment:
This comment is an atavism in this context, right?  Whether we do vectors-of-packable-types or iterator-pairs-of-packable-types or whatever, the whole point of this function is going to be *not* needing standard-types, pre-sized receive buffers, etc.

Reply:
Yep... copy pasta failure